src/libfsm/print/c.c

```diff
@@ -222,10 +222,16 @@ print_case(FILE *f, const struct ir *ir,
 	assert(f != NULL);
 	assert(cs != NULL);
 
+	assert(state_id < ir->n);
+	const struct fsm_state_metadata state_metadata = {
+		.end_ids = ir->states[state_id].endids.ids,
+		.end_id_count = ir->states[state_id].endids.count,
+	};
+
```

Comment:
why not pass whatever type `&ir->states[state_id]..endids` is?

Reply:
`struct ir_state_endids` is defined in an internal `ir.h` header, and an upcoming PR is going to add another pair of fields to the state metadata struct, for eager output IDs and their count.